Remove plexus-container-default depedency

Based on codehaus-plexus/plexus-containers#17 this dependency is obsolete and should be removed.

Why? I am having a problem running mapstruct in intelij when using module-info.java

```
Information:java: compiler message file broken: key=compiler.misc.msg.bug arguments=11, {1}, {2}, {3}, {4}, {5}, {6}, {7}
Information:java: java.lang.module.FindException: Unable to derive module descriptor for /Users/mspiller/.m2/repository/org/codehaus/plexus/plexus-container-default/1.6/plexus-container-default-1.6.jar
Information:java: 	at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:647)
Information:java: 	at java.base/jdk.internal.module.ModulePath.readModule(ModulePath.java:330)
Information:java: 	at java.base/jdk.internal.module.ModulePath.scan(ModulePath.java:236)
Information:java: 	at java.base/jdk.internal.module.ModulePath.scanNextEntry(ModulePath.java:189)
Information:java: 	at java.base/jdk.internal.module.ModulePath.findAll(ModulePath.java:165)
Information:java: 	at java.base/java.lang.module.Resolver.findAll(Resolver.java:842)
Information:java: 	at java.base/java.lang.module.Resolver.bind(Resolver.java:223)
Information:java: 	at java.base/java.lang.module.Configuration.resolveAndBind(Configuration.java:482)
Information:java: 	at java.base/java.lang.module.Configuration.resolveAndBind(Configuration.java:288)
Information:java: 	at jdk.compiler/com.sun.tools.javac.file.JavacFileManager.getServiceLoader(JavacFileManager.java:985)
Information:java: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Information:java: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Information:java: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Information:java: 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
Information:java: 	at org.jetbrains.jps.javac.JavacMain$2.invoke(JavacMain.java:290)
Information:java: 	at com.sun.proxy.$Proxy22.getServiceLoader(Unknown Source)
Information:java: 	at com.sun.proxy.$Proxy23.getServiceLoader(Unknown Source)
Information:java: 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.initProcessorLoader(JavacProcessingEnvironment.java:261)
Information:java: 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.<init>(JavacProcessingEnvironment.java:237)
Information:java: 	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.instance(JavacProcessingEnvironment.java:193)
Information:java: 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.initProcessAnnotations(JavaCompiler.java:1136)
Information:java: 	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:922)
Information:java: 	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
Information:java: 	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
Information:java: 	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
Information:java: 	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:94)
Information:java: 	at org.jetbrains.jps.javac.JavacMain.compile(JavacMain.java:195)
Information:java: 	at org.jetbrains.jps.incremental.java.JavaBuilder.compileJava(JavaBuilder.java:460)
Information:java: 	at org.jetbrains.jps.incremental.java.JavaBuilder.compile(JavaBuilder.java:330)
Information:java: 	at org.jetbrains.jps.incremental.java.JavaBuilder.doBuild(JavaBuilder.java:255)
Information:java: 	at org.jetbrains.jps.incremental.java.JavaBuilder.build(JavaBuilder.java:213)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.runModuleLevelBuilders(IncProjectBuilder.java:1324)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.runBuildersForChunk(IncProjectBuilder.java:1004)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.buildTargetsChunk(IncProjectBuilder.java:1071)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.buildChunkIfAffected(IncProjectBuilder.java:965)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.buildChunks(IncProjectBuilder.java:794)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.runBuild(IncProjectBuilder.java:376)
Information:java: 	at org.jetbrains.jps.incremental.IncProjectBuilder.build(IncProjectBuilder.java:178)
Information:java: 	at org.jetbrains.jps.cmdline.BuildRunner.runBuild(BuildRunner.java:139)
Information:java: 	at org.jetbrains.jps.cmdline.BuildSession.runBuild(BuildSession.java:288)
Information:java: 	at org.jetbrains.jps.cmdline.BuildSession.run(BuildSession.java:121)
Information:java: 	at org.jetbrains.jps.cmdline.BuildMain$MyMessageHandler.lambda$channelRead0$0(BuildMain.java:228)
Information:java: 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
Information:java: 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
Information:java: 	at java.base/java.lang.Thread.run(Thread.java:834)
Information:java: Caused by: java.lang.IllegalArgumentException: plexus.container.default: Invalid module name: 'default' is not a Java identifier
Information:java: 	at java.base/jdk.internal.module.Checks.requireModuleName(Checks.java:59)
Information:java: 	at java.base/java.lang.module.ModuleDescriptor$Builder.<init>(ModuleDescriptor.java:1550)
Information:java: 	at java.base/java.lang.module.ModuleDescriptor.newAutomaticModule(ModuleDescriptor.java:2438)
Information:java: 	at java.base/jdk.internal.module.ModulePath.deriveModuleDescriptor(ModulePath.java:507)
Information:java: 	at java.base/jdk.internal.module.ModulePath.readJar(ModulePath.java:643)
Information:java: 	... 49 more
Information:java: Errors occurred while compiling module 'zzzz-api'
Information:javac 11 was used to compile java sources
Information:2019-08-09 16:09 - Build completed with 1 error and 1 warning in 3 s 245 ms
Error:java: Compilation failed: internal java compiler error
```

--- a/.gitignore
+++ b/.gitignore
@@ -24,3 +24,4 @@ test-output
 # Misc.
 .DS_Store
 checkstyle.cache
+.flattened-pom.xml

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -578,6 +578,11 @@
                         </parameter>
                     </configuration>
                 </plugin>
+                <plugin>
+                    <groupId>org.codehaus.mojo</groupId>
+                    <artifactId>flatten-maven-plugin</artifactId>
+                    <version>1.2.2</version>
+                </plugin>
 
             </plugins>
         </pluginManagement>
@@ -614,6 +619,7 @@
                         <exclude>mvnw*</exclude>
                         <exclude>**/*.asciidoc</exclude>
                         <exclude>**/binding.xjb</exclude>
+                        <exclude>**/*.flattened-pom.xml</exclude>
                     </excludes>
                     <mapping>
                         <java>SLASHSTAR_STYLE</java>
@@ -699,6 +705,34 @@
                             <goal>javadoc-no-fork</goal>
                         </goals>
                         <phase>verify</phase>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>flatten-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <!-- Tidy up all POMs before they are published -->
+                        <id>flatten</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>flatten</goal>
+                        </goals>
+                        <configuration>
+                            <updatePomFile>true</updatePomFile>
+                            <flattenMode>ossrh</flattenMode>
+                            <pomElements>
+                                <parent>expand</parent>
+                            </pomElements>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>flatten-clean</id>
+                        <phase>clean</phase>
+                        <goals>
+                            <goal>clean</goal>
+                        </goals>
                     </execution>
                 </executions>
             </plugin>

--- a/processor/pom.xml
+++ b/processor/pom.xml
@@ -209,6 +209,21 @@
                     </execution>
                 </executions>
             </plugin>
+            <!-- The flatten plugin should run after the shading is done -->
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>flatten-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <!-- Tidy up all POMs before they are published -->
+                        <id>flatten</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>flatten</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-dependency-plugin</artifactId>
@@ -334,6 +349,7 @@
                     <artifactId>jaxb-api</artifactId>
                     <version>2.3.1</version>
                     <scope>provided</scope>
+                    <optional>true</optional>
                 </dependency>
             </dependencies>
         </profile>

